libraries/AP_BattMonitor/AP_BattMonitor_FuelLevel_Reventech.cpp

```diff
@@ -0,0 +1,91 @@
+   MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+   GNU General Public License for more details.
+
+   You should have received a copy of the GNU General Public License
+   along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ */
+
+#include "AP_BattMonitor_config.h"
+
+#if AP_BATTERY_FUELLEVEL_REVENTECH_ENABLED
+
+#include <AP_HAL/AP_HAL.h>
+#include "AP_BattMonitor_FuelLevel_Reventech.h"
+
+/*
+  "battery" monitor for liquid fuel level systems.
+
+  Output is:
+
+    - voltage is the calibrated fuel level in the tank
+ */
+extern const AP_HAL::HAL& hal;
+
+const AP_Param::GroupInfo AP_BattMonitor_FuelLevel_Reventech::var_info[] = {
+
+    // @Param: FL_PIN
+    // @DisplayName: Fuel level analog pin number
+    // @Description: Analog input pin that fuel level sensor is connected to. Airspeed ports can be used for Analog input. When using analog pin 103, the maximum value of the input in 3.3V.
+    // @Values: -1:Not Used,11:Pixracer,13:Pixhawk ADC4,14:Pixhawk ADC3,15:Pixhawk ADC6/Pixhawk2 ADC,103:Pixhawk SBUS,8: F405 Periph ADC Pin
+    AP_GROUPINFO("FL_PIN", 45, AP_BattMonitor_FuelLevel_Reventech, _pin, 8),
+
+    // @Param: LS800_MAX_LEVEL
+    // @DisplayName: LS800 maximum fuel level
+    AP_GROUPINFO("LS800_MAX", 46, AP_BattMonitor_FuelLevel_Reventech, _ls800_max_fuel_level_litres, 8.2),
+
+    AP_GROUPEND
+};
+
+/// Constructor
+AP_BattMonitor_FuelLevel_Reventech::AP_BattMonitor_FuelLevel_Reventech(AP_BattMonitor &mon, AP_BattMonitor::BattMonitor_State &mon_state, AP_BattMonitor_Params &params) :
+    AP_BattMonitor_Backend(mon, mon_state, params)
+{
+    AP_Param::setup_object_defaults(this, var_info);
+    _state.var_info = var_info;
+
+    _analog_source = hal.analogin->channel(_pin);
+}
+
+/*
+  read - read the "voltage" and "current"
+*/
+void AP_BattMonitor_FuelLevel_Reventech::read()
+{
+    if (_analog_source == nullptr) {
+        return;
+    }
+
+    if (!_analog_source->set_pin(_pin)) {
+        _state.healthy = false;
+        return;
+    }
+    _state.healthy = true;
+
+    // get voltage from an ADC pin
+    const float ls800_raw = (_analog_source->read_average() - CPN_ADC_5V_OFF) / CPN_ADC_5V_SLP;
+
+    // Converting sensor reading to actual volume in tank in Litres (quadratic fit)
+    const float voltage = (LS800_FIT_THIRD_COEFF * ls800_raw * ls800_raw * ls800_raw) + (LS800_FIT_SECOND_COEFF * ls800_raw * ls800_raw) + 
+                            (LS800_FIT_FIRST_COEFF * ls800_raw) + LS800_FIT_OFFSET;
+
+    // Add averaging filter at 1Hz for 1minute duration
+    const float filtered_voltage = _ls800_filter.apply(voltage);    
+   
+    // output the ADC voltage to the voltage field for easier calibration of sensors
+    // also output filtered voltage as a measure of tank slosh filtering
+    // this could be useful for tuning the LPF
+
+    _state.voltage = constrain_float(filtered_voltage,0.0,_ls800_max_fuel_level_litres);  
```

Comment:
This seems strange; why do we need a parameter for the max fuel level here?  The sensor reads what the sensor reads....

Reply:
@tridge requested it in the previous PR on this driver. Probably more of a sanity check item if different tanks are used. Though a different tank will require a new calibration fit.

Reply:
gone in current PR